### ISSUE TITLE
Each test should be indepedent with with regard to .config

### ISSUE THREAD

**Issue by thomasfrederikhoeck** [andersbogsnes/ml_tooling#133]
#### Problem description
Using
```python
base.reset_config()

```
in one test affects other tests. This makes it hard to localize the error.
One example is how
```
test_get_default_metric_works_as_expected_with_pipeline
```
affects 
```
test_log_context_manager_logs_when_gridsearching
```
Checklist
---------
- [ ] I have updated ml-tooling to the newest version `pip install ml-tooling --upgrade`
- [ ] I have checked that this issue does not already exist

**Comment by andersbogsnes** [comment]
I've changed the tests to be function scoped now, the performance hit was low enough that it made sense.
Try pulling the new changes and see if it still fails

**Comment by andersbogsnes** [comment]
closed by #132
